### PR TITLE
add core api-gateway-service

### PR DESCRIPTION
**Pull request by AfanasievYuriy**

### CODE REVIEW

File: api-gateway-service/settings.gradle [hunk 1]
```diff
@@ -0,0 +1,6 @@
+pluginManagement {
```
Comment (andrsuh) [inline api-gateway-service/settings.gradle]: you don't need this

File: api-gateway-service/build.gradle [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+sourceCompatibility = 1.8
+
+repositories {
+    mavenCentral()
+    mavenLocal()
+}
+
+dependencyManagement {
+    imports {
+        mavenBom "org.springframework.cloud:spring-cloud-dependencies:Greenwich.SR2"
```
Comment (andrsuh) [inline api-gateway-service/build.gradle]: version should be moved to gradle properties

File: api-gateway-service/build.gradle [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    }
+}
+
+dependencies {
+    compile 'org.springframework.cloud:spring-cloud-starter-eureka:1.4.7.RELEASE'
```
Comment (andrsuh) [inline api-gateway-service/build.gradle]: is you use dependency-management why do you explicitly specify the version of spring-cloud related artifacts? Check if they are presented in BOM and remove versions if they are

File: api-gateway-service/build.gradle [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    compile 'org.springframework.cloud:spring-cloud-starter-zuul:1.4.7.RELEASE'
+    compile 'org.springframework.boot:spring-boot-starter-web'
+    compile 'com.microservices.laundrymanagement.api:api:0.1-SNAPSHOT'
```
Comment (andrsuh) [inline api-gateway-service/build.gradle]: move api versions to gradle.properties

File: api-gateway-service/settings.gradle (hunk 1 above)
Comment (andrsuh) [reply api-gateway-service/settings.gradle]: I think

File: order-management-service/app/build.gradle [hunk 5]
```diff
@@ -32,6 +32,7 @@ dependencyManagement {
 }
 
 dependencies {
+    compile 'org.springframework.cloud:spring-cloud-starter-eureka:1.4.7.RELEASE'
```
Comment (andrsuh) [inline order-management-service/app/build.gradle]: same as above

File: tariff-management-service/app/build.gradle [hunk 6]
```diff
@@ -25,6 +25,7 @@ repositories {
 }
 
 dependencies {
+    compile 'org.springframework.cloud:spring-cloud-starter-eureka:1.4.7.RELEASE'
```
Comment (andrsuh) [inline tariff-management-service/app/build.gradle]: same as above

File: api-gateway-service/build.gradle (hunk 2 above)
Comment (AfanasievYuriy) [reply api-gateway-service/build.gradle]: we should really discuss current state of graddle in our project. I guess i've missed a lot

Comment (AfanasievYuriy) [reply api-gateway-service/build.gradle]: this line doesn't work, so i use versions of eureka and zuul explicitly

File: api-gateway-service/build.gradle (hunk 3 above)
Comment (AfanasievYuriy) [reply api-gateway-service/build.gradle]: answered above

File: api-gateway-service/settings.gradle (hunk 1 above)
Comment (AfanasievYuriy) [reply api-gateway-service/settings.gradle]: as you wish
this was copied from TC

File: api-gateway-service/build.gradle (hunk 2 above)
Comment (andrsuh) [reply api-gateway-service/build.gradle]: I meant there should be a placeholder with release train version

Comment (andrsuh) [reply api-gateway-service/build.gradle]: But you are right We've probably missed a lot with gradle

File: api-gateway-service/build.gradle [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+dependencyManagement {
+    imports {
+        mavenBom "org.springframework.cloud:spring-cloud-dependencies:${releaseTrainVersion}"
+    }
+}
+
+dependencies {
+    compile 'org.springframework.cloud:spring-cloud-starter-netflix-eureka-client'
```
Comment (andrsuh) [inline api-gateway-service/build.gradle:35]: It works without version?

File: order-management-service/app/build.gradle (hunk 5 above)
Comment (andrsuh) [inline order-management-service/app/build.gradle]: can be get rid of version here?

File: order-management-service/app/src/main/resources/application.properties [hunk 8]
```diff
@@ -12,6 +12,9 @@ spring.jpa.hibernate.ddl-auto=validate
 spring.jpa.properties.hibernate.default_schema=${database.schema.name}
 spring.flyway.schemas=${database.schema.name}
 
+#Discovery-service properties
+eureka.client.serviceUrl.defaultZone=http://localhost:8761/eureka/
```
Comment (andrsuh) [inline order-management-service/app/src/main/resources/application.properties:25]: please more comprehensive comment

File: api-gateway-service/build.gradle (hunk 7 above)
Comment (AfanasievYuriy) [reply api-gateway-service/build.gradle:35]: yep

File: order-management-service/app/build.gradle (hunk 5 above)
Comment (AfanasievYuriy) [reply order-management-service/app/build.gradle]: I can move it into properties file not BOM (i won't work with this crap)

**Review by andrsuh — APPROVED** [review]

**Pull request merged**